apps/api/src/app/workflows-v2/usecases/generate-preview/generate-preview.usecase.ts

```diff
@@ -309,3 +321,106 @@ class FrameworkError {
   message: string;
   name: string;
 }
+
+function sanitizeControlValuesByOutputSchema(
+  controlValues: Record<string, unknown>,
+  type: StepTypeEnum
+): Record<string, unknown> {
+  const outputSchema = channelStepSchemas[type].output || actionStepSchemas[type].output;
```

Comment:
💯 Nice, reusing schema definitions from @novu/framework!

Reply:
haha thanks, i thought it was an obvious move here ✨